### ISSUE TITLE
remote write 2.0 - test for agent mode with metadata-wal-records and type-and-unit features

### ISSUE THREAD

**Issue by cstyan** [prometheus/prometheus#13483]
just need to confirm here that this config option is/can be passed to proemtheus in agent mode correctly and that metadata will be properly appended to the WAL

this should be the case already but we should confirm

**Comment by bwplotka** [comment]
I would say, let's also check https://github.com/prometheus/prometheus/issues/16610 not sure about the agent mode path anymore..

**Comment by cstyan** [comment]
yeah it seems like there's some fairly large agent mode users but agent mode itself has no active owner ATM so I think it's a good idea to double check this and your issue

**Comment by bwplotka** [comment]
Given we deprio the metadata-wal-records and we noticed that for `type-and`unit` there's not much to test in agent (everything is done on scrape), we can mark this as done.
